srcs/go/kungfu/workspace.go

```diff
@@ -39,3 +41,25 @@ func (w Workspace) split(p partitionFunc, k int) []Workspace {
 	}
 	return ws
 }
+
+type shardHashFunc func(int, string) uint64
```

Comment:
We need to add comments to this function.

``A strategyHashFunc is to map a given tensor to a communication strategy. KungFu can create parallel communication strategy to balance the workload in the network core and edge links. We support hash the tensors based on their names or their ID if they are chunked.``